### PR TITLE
fix:[CDS-43763]:Deployment Template Creation

### PR DESCRIPTION
**Pull request by namanharness**
### Summary

Main Changes:-
- Added Cypress Automation for Deployment Template Creation.

Other Changes:-
1.)Improved StepTemplateSchema.spec.ts file
2.) Added a Cypress command to assert visibility of child elements whose parent has conflictiing CSS property.

#### Screenshots
https://user-images.githubusercontent.com/110456765/193250082-a5c66dfe-bd30-4d24-bdb5-f6084140abe5.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11475)

**Comment by namanharness** [comment]
retrigger jest

File: cypress/integration/72-templates-library/DeploymentTemplateCreate.spec.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    })
+    cy.intercept('POST', templateCreationCall, incompleteTemplateCreationResponse).as('templateCreation')
+    cy.intercept('POST', templateListCallAfterSelection, stepTemplateListCallAfterSelectionResponse).as('stepListCall')
+    cy.intercept('GET', templateInputsCallAfterSelection, deploymentTemplateInputCallAfterSelectionResponse).as(
+      'Step Template Input'
+    )
+    cy.visitPageAssertion('[class*=TemplatesPage-module_templatesPageBody]')
+
+    cy.contains('span', 'New Template').click()
+    cy.get('.bp3-menu > :nth-child(4)').click()
```
Comment (PratyushGar) [inline cypress/integration/72-templates-library/DeploymentTemplateCreate.spec.ts:68]: Can we assert with name instead, order is subject to change

Comment (namanharness) [reply cypress/integration/72-templates-library/DeploymentTemplateCreate.spec.ts]: done

**Review by PratyushGar — DISMISSED** [review]

**Review by samarth-sharma31 — DISMISSED** [review]

**Comment by namanharness** [comment]
retrigger jest

**Review by samarth-sharma31 — DISMISSED** [review]

**Comment by namanharness** [comment]
retrigger jest

**Review by PratyushGar — APPROVED** [review]

**Review by samarth-sharma31 — APPROVED** [review]

**Pull request merged**